src/main/java/techreborn/blockentity/machine/tier2/MinerBlockEntity.java

```diff
@@ -0,0 +1,227 @@
+package techreborn.blockentity.machine.tier2;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+import reborncore.common.blockentity.MachineBaseBlockEntity;
+import reborncore.common.screen.BuiltScreenHandler;
+import reborncore.common.screen.BuiltScreenHandlerProvider;
+import reborncore.common.screen.builder.ScreenHandlerBuilder;
+import reborncore.common.util.RebornInventory;
+import techreborn.blockentity.machine.GenericMachineBlockEntity;
+import techreborn.blockentity.machine.tier0.block.BlockProcessable;
+import techreborn.init.TRBlockEntities;
+import techreborn.init.TRContent;
+
+import java.util.Iterator;
+import java.util.Set;
+
+public class MinerBlockEntity extends GenericMachineBlockEntity implements BuiltScreenHandlerProvider, BlockProcessable {
+
+	/**
+	 * Horizontal manhattan distance the ore prospection can cover per level.
+	 */
+	static final int DEFAULT_PROSPECTING_RANGE = 2;
+
+	// Vertical distance the mining pipe can be extended from the miner
+	static final int DEFAULT_DEPTH = 0;
+
+	// Blocks per tick that are examined when a vertical level is scanned for ore.
+	static final float PROSPECTING_SPEED = 1f;
+
+	static final int MINING_TOOL_SLOT = 0;
+	static final int PROSPECTING_TOOL_SLOT = 1;
+
+	static final int MINING_PIPE_SLOT = 2;
+	static final int OUTPUT_SLOT = 3;
+	static final int ENERGY_SLOT = 4;
+
+	static final Set<Item> ALLOWED_MINING_TOOLS = Set.of(TRContent.BASIC_DRILL, TRContent.ADVANCED_DRILL, TRContent.INDUSTRIAL_DRILL);
```

Comment:
Maybe it is better to use tags here

Reply:
Should I just define it in resources/data or add to datagen?